src/Symfony/Component/RateLimiter/FixedWindowLimiter.php

```diff
@@ -53,18 +55,40 @@ public function consume(int $tokens = 1): bool
                 $window = new Window($this->id, $this->interval);
             }
 
-            $hitCount = $window->getHitCount();
-            $availableTokens = $this->limit - $hitCount;
+            $availableTokens = $this->getAvailableTokens($window);
+            $windowStart = \DateTimeImmutable::createFromFormat('U', time());
             if ($availableTokens < $tokens) {
-                return false;
+                return new Limit($availableTokens, $this->getRetryAfter($windowStart), false, $this->strategy, [
+                    'limit' => $this->limit,
+                    'interval' => $this->interval,
+                    'id' => $this->id,
+                ]);
             }
 
             $window->add($tokens);
             $this->storage->save($window);
 
-            return true;
+            return new Limit($this->getAvailableTokens($window), $this->getRetryAfter($windowStart), true, $this->strategy, [
+                'limit' => $this->limit,
+                'interval' => $this->interval,
+                'id' => $this->id,
+            ]);
         } finally {
             $this->lock->release();
         }
     }
+
+    private function getRetryAfter(\DateTimeImmutable $windowStart)
+    {
+        return $windowStart->add(
+            new \DateInterval(sprintf('PT%sS', $this->interval))
+        );
```

Comment:
```suggestion
        return $windowStart->add(new \DateInterval(sprintf('PT%sS', $this->interval)));
```

(in Symfony, the general view is that common screens are wider than they are longer, so it's "better" use of screen space if a line is longer unless it becomes unreadable)